auditbeat/module/file_integrity/kprobes/verifier.go

```diff
@@ -0,0 +1,232 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build linux
+
+package kprobes
+
+import (
+	"bytes"
+	"context"
+	"embed"
+	"errors"
+	"io/fs"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/elastic/beats/v7/auditbeat/tracing"
+
+	tkbtf "github.com/elastic/tk-btf"
+)
+
+//go:embed embed
+var embedBTFFolder embed.FS
+
+func getVerifiedProbes(ctx context.Context, timeout time.Duration) (map[tracing.Probe]tracing.AllocateFn, executor, error) {
+
+	fExec := newFixedThreadExecutor(ctx)
+
+	probeMgr, err := newProbeManager(fExec)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	specs, err := loadAllSpecs()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var allErr error
+	for len(specs) > 0 {
```

Comment:
any reason you can't loop on `specs` normally instead of advancing manually?

Reply:
there are cases that we need to re-run with a specific spec such as when we adjust the filters of the fsnotify probes look at `OnErr`